packages/flutter/lib/src/material/navigation_bar_theme.dart

```diff
@@ -0,0 +1,221 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material_state.dart';
+import 'navigation_bar.dart';
+import 'theme.dart';
+
+/// Defines default property values for descendant [NavigationBar]
+/// widgets.
+///
+/// Descendant widgets obtain the current [NavigationBarThemeData] object
+/// using `NavigationBarTheme.of(context)`. Instances of
+/// [NavigationBarThemeData] can be customized with
+/// [NavigationBarThemeData.copyWith].
+///
+/// Typically a [NavigationBarThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.navigationBarTheme].
+///
+/// All [NavigationBarThemeData] properties are `null` by default.
+/// When null, the [NavigationBar] will use the values from [ThemeData]
+/// if they exist, otherwise it will provide its own defaults based on the
+/// overall [Theme]'s textTheme and colorScheme. See the individual
+/// [NavigationBar] properties for details.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class NavigationBarThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.NavigationBarTheme].
+  const NavigationBarThemeData({
+    this.height,
+    this.backgroundColor,
+    this.indicatorColor,
+    this.labelTextStyle,
+    this.iconTheme,
+    this.labelBehavior,
+  });
+
+  /// The height of the [NavigationBar].
+  final double? height;
```

Comment:
Each of these properties should say something like "overrides the default value of [NavigationBar.height]". Following the [NavigationBar.height] link should reveal the default as well as the complete override logic (I believe that's true already).

Reply:
Done